src/components/BookingModal/BookingModalForm/index.js

```diff
@@ -24,29 +25,32 @@ const BookingModalForm = props => {
     booking,
     isSameDay,
   } = props;
-
+  const { eventStatus: {eventStatusId} } = booking;
+  const isEventCancelled = eventStatusId === holidayStatus.CANCELLED;
+  const rejectionMessages = booking.messages && !isEventCancelled;
   const { eventTypeId } = formData;
   const buttonTextValue =
   `${eventTypeId !== eventTypes.ANNUAL_LEAVE ? '' : bookingDuration === 0.5 ? 'Half' : bookingDuration}
   ${eventTypeId !== eventTypes.ANNUAL_LEAVE ? 'WFH' : bookingDuration > 1 ? 'Days' : 'Day'}`;
 
   const createCtas = () => {
+    const isButtonDisabled = isEventCancelled ? isEventCancelled : !formIsValid;
```

Comment:
In the other refactor branch, TS-109, there is now a function called `isValidBooking` which drives the logic for this, as there are quite a few different conditions where the button should be disabled. I recommend merging that branch into this one and using it.

Reply:
Emmm maybe wait until that other branch has made it into develop and merge from there just to keep each ticket's changes separate.

Reply:
^